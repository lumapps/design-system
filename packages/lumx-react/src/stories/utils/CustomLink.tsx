/* eslint-disable jsx-a11y/anchor-has-content,jsx-a11y/anchor-is-valid */
import React from 'react';

/**
 * Example custom link that can be used in linkAs props.
 */
export const CustomLink: any = ({ ...props }) => <a href="#" data-custom-link {...props} />;
