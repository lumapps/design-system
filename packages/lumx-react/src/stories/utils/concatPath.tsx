export type PathPart = string | number;
export type Path = PathPart | Array<PathPart>;

/**
 * Concat flatten object path
 *
 * @example concatPath('foo', 'bar') // => 'foo.bar'
 * @example concatPath(['foo', 0]) // => 'foo[0]'
 * @example concatPath('foo', 0, ['bar']) // => 'foo[0].bar'
 */
export const concatPath = (...prefix: Path[]) => {
    const [first, ...rest] = prefix.flat();
    return rest.reduce<string>((acc, part) => {
        if (typeof part === 'number') return `${acc}[${part}]`;
        return `${acc}.${part}`;
    }, String(first));
};
