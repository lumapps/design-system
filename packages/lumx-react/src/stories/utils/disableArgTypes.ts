export function disableArgTypes(argTypes: string[]) {
    return Object.fromEntries(argTypes.map((argType) => [argType, { table: { disable: true } }]));
}
