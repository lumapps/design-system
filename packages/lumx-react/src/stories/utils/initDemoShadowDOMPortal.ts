/** Init a demo shadow DOM to use as portal container */
export const initDemoShadowDOMPortal = () => {
    const div = document.createElement('div');
    document.body.appendChild(div);
    const container = div.attachShadow({ mode: 'open' });
    const style = document.createElement('style');
    style.innerText = `button { color: red; }`;
    container.appendChild(style);
    return { container, teardown: () => div.remove() };
};
