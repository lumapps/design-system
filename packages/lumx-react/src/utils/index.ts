/**
 * WARNING: All modules exported here are exposed to NPM in '@lumx/react/utils'.
 */

export { ClickAwayProvider } from './ClickAwayProvider';
export { Portal, type PortalProps, type PortalInit, PortalProvider, type PortalProviderProps } from './Portal';
