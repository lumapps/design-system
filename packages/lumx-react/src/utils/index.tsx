export * from '@lumx/core/js/utils';

export * from './getRootClassName';

export * from './htmlDecode';

export * from './partitionMulti';

export * from './type';
