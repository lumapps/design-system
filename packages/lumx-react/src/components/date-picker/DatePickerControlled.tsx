import React, { forwardRef } from 'react';
import moment from 'moment';
import classNames from 'classnames';
import { DatePickerProps, Emphasis, IconButton, Toolbar } from '@lumx/react';
import { mdiChevronLeft, mdiChevronRight } from '@lumx/icons';
import { getAnnotatedMonthCalendar, getWeekDays } from '@lumx/core/js/date-picker';
import { Comp } from '@lumx/react/utils';
import { CLASSNAME } from './constants';

/**
 * Defines the props of the component.
 */
export interface DatePickerControlledProps extends DatePickerProps {
    /** Selected month to display. */
    selectedMonth: Date;
    /** On previous month change callback. */
    onPrevMonthChange(): void;
    /** On next month change callback. */
    onNextMonthChange(): void;
}

/**
 * Component display name.
 */
const COMPONENT_NAME = 'DatePickerControlled';

/**
 * DatePickerControlled component.
 *
 * @param  props Component props.
 * @param  ref   Component ref.
 * @return React element.
 */
export const DatePickerControlled: Comp<DatePickerControlledProps, HTMLDivElement> = forwardRef((props, ref) => {
    const {
        locale,
        maxDate,
        minDate,
        nextButtonProps,
        onChange,
        onNextMonthChange,
        onPrevMonthChange,
        previousButtonProps,
        selectedMonth,
        todayOrSelectedDateRef,
        value,
    } = props;
    const days = React.useMemo(() => {
        return getAnnotatedMonthCalendar(locale, minDate, maxDate, moment(selectedMonth));
    }, [locale, minDate, maxDate, selectedMonth]);

    const weekDays = React.useMemo(() => {
        return getWeekDays(locale);
    }, [locale]);

    return (
        <div ref={ref} className={`${CLASSNAME}`}>
            <Toolbar
                className={`${CLASSNAME}__toolbar`}
                after={
                    <IconButton
                        {...nextButtonProps}
                        emphasis={Emphasis.low}
                        icon={mdiChevronRight}
                        onClick={onNextMonthChange}
                    />
                }
                before={
                    <IconButton
                        {...previousButtonProps}
                        emphasis={Emphasis.low}
                        icon={mdiChevronLeft}
                        onClick={onPrevMonthChange}
                    />
                }
                label={
                    <span className={`${CLASSNAME}__month`}>
                        {moment(selectedMonth).locale(locale).format('MMMM YYYY')}
                    </span>
                }
            />
            <div className={`${CLASSNAME}__calendar`}>
                <div className={`${CLASSNAME}__week-days ${CLASSNAME}__days-wrapper`}>
                    {weekDays.map((weekDay) => (
                        <div key={weekDay.unix()} className={`${CLASSNAME}__day-wrapper`}>
                            <span className={`${CLASSNAME}__week-day`}>
                                {weekDay.format('dddd').slice(0, 1).toLocaleUpperCase()}
                            </span>
                        </div>
                    ))}
                </div>

                <div className={`${CLASSNAME}__month-days ${CLASSNAME}__days-wrapper`}>
                    {days.map((annotatedDate) => {
                        if (annotatedDate.isDisplayed) {
                            return (
                                <div key={annotatedDate.date.unix()} className={`${CLASSNAME}__day-wrapper`}>
                                    <button
                                        ref={
                                            (value && annotatedDate.date.isSame(value, 'day')) ||
                                            (!value && annotatedDate.isToday)
                                                ? todayOrSelectedDateRef
                                                : null
                                        }
                                        className={classNames(`${CLASSNAME}__month-day`, {
                                            [`${CLASSNAME}__month-day--is-selected`]:
                                                value && annotatedDate.date.isSame(value, 'day'),
                                            [`${CLASSNAME}__month-day--is-today`]:
                                                annotatedDate.isClickable && annotatedDate.isToday,
                                        })}
                                        disabled={!annotatedDate.isClickable}
                                        type="button"
                                        onClick={() => onChange(moment(annotatedDate.date).toDate())}
                                    >
                                        <span>{annotatedDate.date.format('DD')}</span>
                                    </button>
                                </div>
                            );
                        }
                        return <div key={annotatedDate.date.unix()} className={`${CLASSNAME}__day-wrapper`} />;
                    })}
                </div>
            </div>
        </div>
    );
});
DatePickerControlled.displayName = COMPONENT_NAME;
DatePickerControlled.className = CLASSNAME;
