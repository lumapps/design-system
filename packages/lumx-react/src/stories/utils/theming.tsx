import React from 'react';
import { Predicate } from '@lumx/react/utils/type';
import { FlexBox } from '@lumx/react';

type Property = string | { value: string };
interface PropertyTree {
    [key: string]: PropertyTree | Property;
}
type Entry = [key: string, value: string];
type Variables = Record<string, string>;

function asValue(prop?: PropertyTree | Property): string | undefined {
    if (typeof prop === 'string') {
        return prop;
    }
    if (typeof prop === 'object' && 'value' in prop) {
        return prop.value as string;
    }
    return undefined;
}

/** Convert property tree into a list of args entries ([key, value]) */
function* toArgEntries(props: PropertyTree | Property, parent = ''): Iterable<Entry> {
    const value = asValue(props);
    if (value) {
        yield [parent, value];
    } else {
        for (const [key, prop] of Object.entries(props)) {
            const newParent = parent ? `${parent}-${key}` : key;
            yield* toArgEntries(prop, newParent);
        }
    }
}
function* parseVariableEntries(cssString: string, strict = false) {
    for (const line of cssString.split(/(?<=;)/)) {
        if (!line.trim()) continue;
        const match = line.match(/^\s*(--lumx[^:]+)\s*:\s*([^;]+);/m);
        if (!match) {
            if (strict) throw new Error(`Could not parse \`${line}\``);
            else continue;
        }
        const [, variableName, variableValue] = match;
        yield [variableName, variableValue];
    }
}

function VariableEditor({
    variable: [key, value],
    setVariable,
    defaultStyle,
}: {
    variable: Entry;
    setVariable: (newVariable: Entry) => void;
    defaultStyle: Variables;
}) {
    const onChange = React.useCallback<React.ChangeEventHandler<HTMLInputElement>>(
        (evt) => {
            setVariable([evt.target.name, evt.target.value]);
        },
        [setVariable],
    );

    return (
        <FlexBox as="label" orientation="horizontal" hAlign="center" style={{ whiteSpace: 'nowrap' }}>
            <span style={{ flexShrink: 0 }}>{key}</span>:
            <input
                name={key}
                onChange={onChange}
                style={{ border: 0, padding: 8, fieldSizing: 'content', display: 'inline' } as any}
                value={defaultStyle[key] === value ? '' : value}
                placeholder={value}
            />
            ;
        </FlexBox>
    );
}

function VariablesEditor({
    style,
    defaultStyle,
    setStyle,
}: {
    style: Variables;
    defaultStyle: Variables;
    setStyle: (fn: (newStyle: Variables) => Variables) => void;
}) {
    const setVariable = React.useCallback(
        ([key, value]: Entry) => {
            setStyle((prevStyle) => ({ ...prevStyle, [key]: value }));
        },
        [setStyle],
    );
    return (
        <details style={{ fontFamily: 'monospace' }} className="lumx-spacing-margin-vertical-huge">
            <summary className="lumx-typography-subtitle2">Edit variables</summary>
            {Object.entries(style).map((variable) => (
                <VariableEditor
                    key={variable[0]}
                    defaultStyle={defaultStyle}
                    variable={variable}
                    setVariable={setVariable}
                />
            ))}
        </details>
    );
}

const FILTER_PROPS = ['theme', 'emphasis', 'size'];

export function withTheming<P extends PropertyTree>({
    properties,
    values,
}: {
    properties?: P;
    values?: string | Partial<P>;
}) {
    const initialStyle = values
        ? Object.fromEntries(
              typeof values === 'string'
                  ? parseVariableEntries(values)
                  : toArgEntries(values as PropertyTree, '--lumx'),
          )
        : [];
    const defaultStyle = Object.fromEntries(properties ? Array.from(toArgEntries(properties, '--lumx')) : []);

    const varNames = new Set(Object.keys(initialStyle).concat(Object.keys(defaultStyle)));

    function filteredStyles(args: any) {
        // Create predicate
        const filterArgs = Object.entries(args)
            .map(([argName, argValue]) => {
                if (typeof argValue === 'string' && FILTER_PROPS.includes(argName)) {
                    return (varName: string) =>
                        // Has the given arg with value
                        varName.match(new RegExp(`${argName}-${argValue}`)) ||
                        (argName === 'emphasis' && argValue === 'medium' && varName.match(/emphasis-selected/)) ||
                        // Or does not have the arg at all
                        !varName.match(new RegExp(`${argName}-\\w+`));
                }
                return false;
            })
            .filter(Boolean) as Array<Predicate<string>>;
        const styles = Object.fromEntries(
            Array.from(varNames.values())
                .map((varName) => {
                    if (!filterArgs.every((filter) => filter(varName))) return null as any;
                    return [varName, initialStyle[varName] || defaultStyle[varName]];
                })
                .filter(Boolean),
        );
        return () => styles;
    }

    return (Story: any, ctx: any) => {
        const [style, setStyle] = React.useState<Variables>(filteredStyles(ctx.args));

        return (
            <>
                <Story args={{ ...ctx.args, style: { ...ctx.args.style, ...style } }} />
                <VariablesEditor style={style} defaultStyle={defaultStyle} setStyle={setStyle} />
            </>
        );
    };
}
