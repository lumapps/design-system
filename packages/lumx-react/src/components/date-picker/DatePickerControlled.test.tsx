import React, { ReactElement } from 'react';

import { mount, shallow } from 'enzyme';
import 'jest-enzyme';

import { Wrapper } from '@lumx/react/testing/utils';

import { DatePickerControlled, DatePickerControlledProps } from './DatePickerControlled';

const mockedDate = new Date(
    new Date(1487721600).toLocaleString('en-US', {
        timeZone: 'America/Toronto',
    }),
);
Date.now = jest.fn(() => mockedDate.valueOf());

type SetupProps = Partial<DatePickerControlledProps>;

const setup = ({ ...propsOverride }: SetupProps = {}, shallowRendering = true) => {
    const props: DatePickerControlledProps = {
        locale: 'fr',
        onChange: jest.fn(),
        onNextMonthChange: jest.fn(),
        onPrevMonthChange: jest.fn(),
        selectedMonth: mockedDate,
        value: mockedDate,
        nextButtonProps: { label: 'Next month' },
        previousButtonProps: { label: 'Previous month' },
        ...propsOverride,
    };
    const renderer: (el: ReactElement) => Wrapper = shallowRendering ? shallow : mount;
    const wrapper: Wrapper = renderer(<DatePickerControlled {...props} />);

    return { props, wrapper };
};

describe(`<${DatePickerControlled.displayName}>`, () => {
    describe('Snapshots and structure', () => {
        it('should render correctly', () => {
            const { wrapper } = setup();
            expect(wrapper).toMatchSnapshot();
        });
    });
});
