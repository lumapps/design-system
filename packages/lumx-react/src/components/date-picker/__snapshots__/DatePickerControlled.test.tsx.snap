// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<DatePickerControlled> Snapshots and structure should render correctly 1`] = `
<div
  className="lumx-date-picker"
>
  <Toolbar
    after={
      <IconButton
        emphasis="low"
        icon="M8.59,16.58L13.17,12L8.59,7.41L10,6L16,12L10,18L8.59,16.58Z"
        label="Next month"
        onClick={[MockFunction]}
        size="m"
        theme="light"
      />
    }
    before={
      <IconButton
        emphasis="low"
        icon="M15.41,16.58L10.83,12L15.41,7.41L14,6L8,12L14,18L15.41,16.58Z"
        label="Previous month"
        onClick={[MockFunction]}
        size="m"
        theme="light"
      />
    }
    className="lumx-date-picker__toolbar"
    label={
      <span
        className="lumx-date-picker__month"
      >
        janvier 1970
      </span>
    }
  />
  <div
    className="lumx-date-picker__calendar"
  >
    <div
      className="lumx-date-picker__week-days lumx-date-picker__days-wrapper"
    >
      <div
        className="lumx-date-picker__day-wrapper"
        key="947721"
      >
        <span
          className="lumx-date-picker__week-day"
        >
          L
        </span>
      </div>
      <div
        className="lumx-date-picker__day-wrapper"
        key="1034121"
      >
        <span
          className="lumx-date-picker__week-day"
        >
          M
        </span>
      </div>
      <div
        className="lumx-date-picker__day-wrapper"
        key="1120521"
      >
        <span
          className="lumx-date-picker__week-day"
        >
          M
        </span>
      </div>
      <div
        className="lumx-date-picker__day-wrapper"
        key="1206921"
      >
        <span
          className="lumx-date-picker__week-day"
        >
          J
        </span>
      </div>
      <div
        className="lumx-date-picker__day-wrapper"
        key="1293321"
      >
        <span
          className="lumx-date-picker__week-day"
        >
          V
        </span>
      </div>
      <div
        className="lumx-date-picker__day-wrapper"
        key="1379721"
      >
        <span
          className="lumx-date-picker__week-day"
        >
          S
        </span>
      </div>
      <div
        className="lumx-date-picker__day-wrapper"
        key="1466121"
      >
        <span
          className="lumx-date-picker__week-day"
        >
          D
        </span>
      </div>
    </div>
    <div
      className="lumx-date-picker__month-days lumx-date-picker__days-wrapper"
    >
      <div
        className="lumx-date-picker__day-wrapper"
        key="-262800"
      />
      <div
        className="lumx-date-picker__day-wrapper"
        key="-176400"
      />
      <div
        className="lumx-date-picker__day-wrapper"
        key="-90000"
      />
      <div
        className="lumx-date-picker__day-wrapper"
        key="-3600"
      >
        <button
          className="lumx-date-picker__month-day"
          disabled={false}
          onClick={[Function]}
          type="button"
        >
          <span>
            01
          </span>
        </button>
      </div>
      <div
        className="lumx-date-picker__day-wrapper"
        key="82800"
      >
        <button
          className="lumx-date-picker__month-day"
          disabled={false}
          onClick={[Function]}
          type="button"
        >
          <span>
            02
          </span>
        </button>
      </div>
      <div
        className="lumx-date-picker__day-wrapper"
        key="169200"
      >
        <button
          className="lumx-date-picker__month-day"
          disabled={false}
          onClick={[Function]}
          type="button"
        >
          <span>
            03
          </span>
        </button>
      </div>
      <div
        className="lumx-date-picker__day-wrapper"
        key="255600"
      >
        <button
          className="lumx-date-picker__month-day"
          disabled={false}
          onClick={[Function]}
          type="button"
        >
          <span>
            04
          </span>
        </button>
      </div>
      <div
        className="lumx-date-picker__day-wrapper"
        key="342000"
      >
        <button
          className="lumx-date-picker__month-day"
          disabled={false}
          onClick={[Function]}
          type="button"
        >
          <span>
            05
          </span>
        </button>
      </div>
      <div
        className="lumx-date-picker__day-wrapper"
        key="428400"
      >
        <button
          className="lumx-date-picker__month-day"
          disabled={false}
          onClick={[Function]}
          type="button"
        >
          <span>
            06
          </span>
        </button>
      </div>
      <div
        className="lumx-date-picker__day-wrapper"
        key="514800"
      >
        <button
          className="lumx-date-picker__month-day"
          disabled={false}
          onClick={[Function]}
          type="button"
        >
          <span>
            07
          </span>
        </button>
      </div>
      <div
        className="lumx-date-picker__day-wrapper"
        key="601200"
      >
        <button
          className="lumx-date-picker__month-day"
          disabled={false}
          onClick={[Function]}
          type="button"
        >
          <span>
            08
          </span>
        </button>
      </div>
      <div
        className="lumx-date-picker__day-wrapper"
        key="687600"
      >
        <button
          className="lumx-date-picker__month-day"
          disabled={false}
          onClick={[Function]}
          type="button"
        >
          <span>
            09
          </span>
        </button>
      </div>
      <div
        className="lumx-date-picker__day-wrapper"
        key="774000"
      >
        <button
          className="lumx-date-picker__month-day"
          disabled={false}
          onClick={[Function]}
          type="button"
        >
          <span>
            10
          </span>
        </button>
      </div>
      <div
        className="lumx-date-picker__day-wrapper"
        key="860400"
      >
        <button
          className="lumx-date-picker__month-day"
          disabled={false}
          onClick={[Function]}
          type="button"
        >
          <span>
            11
          </span>
        </button>
      </div>
      <div
        className="lumx-date-picker__day-wrapper"
        key="946800"
      >
        <button
          className="lumx-date-picker__month-day"
          disabled={false}
          onClick={[Function]}
          type="button"
        >
          <span>
            12
          </span>
        </button>
      </div>
      <div
        className="lumx-date-picker__day-wrapper"
        key="1033200"
      >
        <button
          className="lumx-date-picker__month-day"
          disabled={false}
          onClick={[Function]}
          type="button"
        >
          <span>
            13
          </span>
        </button>
      </div>
      <div
        className="lumx-date-picker__day-wrapper"
        key="1119600"
      >
        <button
          className="lumx-date-picker__month-day"
          disabled={false}
          onClick={[Function]}
          type="button"
        >
          <span>
            14
          </span>
        </button>
      </div>
      <div
        className="lumx-date-picker__day-wrapper"
        key="1206000"
      >
        <button
          className="lumx-date-picker__month-day"
          disabled={false}
          onClick={[Function]}
          type="button"
        >
          <span>
            15
          </span>
        </button>
      </div>
      <div
        className="lumx-date-picker__day-wrapper"
        key="1292400"
      >
        <button
          className="lumx-date-picker__month-day"
          disabled={false}
          onClick={[Function]}
          type="button"
        >
          <span>
            16
          </span>
        </button>
      </div>
      <div
        className="lumx-date-picker__day-wrapper"
        key="1378800"
      >
        <button
          className="lumx-date-picker__month-day"
          disabled={false}
          onClick={[Function]}
          type="button"
        >
          <span>
            17
          </span>
        </button>
      </div>
      <div
        className="lumx-date-picker__day-wrapper"
        key="1465200"
      >
        <button
          className="lumx-date-picker__month-day lumx-date-picker__month-day--is-selected lumx-date-picker__month-day--is-today"
          disabled={false}
          onClick={[Function]}
          type="button"
        >
          <span>
            18
          </span>
        </button>
      </div>
      <div
        className="lumx-date-picker__day-wrapper"
        key="1551600"
      >
        <button
          className="lumx-date-picker__month-day"
          disabled={false}
          onClick={[Function]}
          type="button"
        >
          <span>
            19
          </span>
        </button>
      </div>
      <div
        className="lumx-date-picker__day-wrapper"
        key="1638000"
      >
        <button
          className="lumx-date-picker__month-day"
          disabled={false}
          onClick={[Function]}
          type="button"
        >
          <span>
            20
          </span>
        </button>
      </div>
      <div
        className="lumx-date-picker__day-wrapper"
        key="1724400"
      >
        <button
          className="lumx-date-picker__month-day"
          disabled={false}
          onClick={[Function]}
          type="button"
        >
          <span>
            21
          </span>
        </button>
      </div>
      <div
        className="lumx-date-picker__day-wrapper"
        key="1810800"
      >
        <button
          className="lumx-date-picker__month-day"
          disabled={false}
          onClick={[Function]}
          type="button"
        >
          <span>
            22
          </span>
        </button>
      </div>
      <div
        className="lumx-date-picker__day-wrapper"
        key="1897200"
      >
        <button
          className="lumx-date-picker__month-day"
          disabled={false}
          onClick={[Function]}
          type="button"
        >
          <span>
            23
          </span>
        </button>
      </div>
      <div
        className="lumx-date-picker__day-wrapper"
        key="1983600"
      >
        <button
          className="lumx-date-picker__month-day"
          disabled={false}
          onClick={[Function]}
          type="button"
        >
          <span>
            24
          </span>
        </button>
      </div>
      <div
        className="lumx-date-picker__day-wrapper"
        key="2070000"
      >
        <button
          className="lumx-date-picker__month-day"
          disabled={false}
          onClick={[Function]}
          type="button"
        >
          <span>
            25
          </span>
        </button>
      </div>
      <div
        className="lumx-date-picker__day-wrapper"
        key="2156400"
      >
        <button
          className="lumx-date-picker__month-day"
          disabled={false}
          onClick={[Function]}
          type="button"
        >
          <span>
            26
          </span>
        </button>
      </div>
      <div
        className="lumx-date-picker__day-wrapper"
        key="2242800"
      >
        <button
          className="lumx-date-picker__month-day"
          disabled={false}
          onClick={[Function]}
          type="button"
        >
          <span>
            27
          </span>
        </button>
      </div>
      <div
        className="lumx-date-picker__day-wrapper"
        key="2329200"
      >
        <button
          className="lumx-date-picker__month-day"
          disabled={false}
          onClick={[Function]}
          type="button"
        >
          <span>
            28
          </span>
        </button>
      </div>
      <div
        className="lumx-date-picker__day-wrapper"
        key="2415600"
      >
        <button
          className="lumx-date-picker__month-day"
          disabled={false}
          onClick={[Function]}
          type="button"
        >
          <span>
            29
          </span>
        </button>
      </div>
      <div
        className="lumx-date-picker__day-wrapper"
        key="2502000"
      >
        <button
          className="lumx-date-picker__month-day"
          disabled={false}
          onClick={[Function]}
          type="button"
        >
          <span>
            30
          </span>
        </button>
      </div>
      <div
        className="lumx-date-picker__day-wrapper"
        key="2588400"
      >
        <button
          className="lumx-date-picker__month-day"
          disabled={false}
          onClick={[Function]}
          type="button"
        >
          <span>
            31
          </span>
        </button>
      </div>
    </div>
  </div>
</div>
`;
